src/Compilers/CSharp/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.cs

```diff
@@ -1305,13 +1369,51 @@ public void SignModuleKeyFileCmdLine()
         {
             string s = "public class C {}";
 
-            var other = CreateStandardCompilation(s, options: TestOptions.ReleaseModule.WithCryptoKeyFile(s_keyPairFile).WithStrongNameProvider(s_defaultProvider));
+            var options = TestOptions.ReleaseModule.WithCryptoKeyFile(s_keyPairFile).WithStrongNameProvider(s_defaultPortableProvider);
+            var other = CreateStandardCompilation(s, options: options);
+
+            var outStrm = new MemoryStream();
+            var success = other.Emit(outStrm);
+            Assert.True(success.Success);
+
+            ConfirmModuleAttributePresentAndAddingToAssemblyResultsInSignedOutput(outStrm, AttributeDescription.AssemblyKeyFileAttribute, legacyStrongName: false);
+        }
+
+        [WorkItem(531195, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/531195")]
+        [Fact()]
+        public void SignRefAssemblyKeyFileCmdLine()
+        {
+            string s = "public class C {}";
+
+            var options = TestOptions.DebugDll.WithCryptoKeyFile(s_keyPairFile).WithStrongNameProvider(s_defaultPortableProvider);
+            var other = CreateStandardCompilation(s, options: options);
+
+            var outStrm = new MemoryStream();
+            var refStrm = new MemoryStream();
+            var success = other.Emit(outStrm, metadataPEStream: refStrm);
+            Assert.True(success.Success);
+
+            outStrm.Position = 0;
+            refStrm.Position = 0;
+
+            Assert.True(IsStreamSigned(outStrm));
+            Assert.True(IsStreamSigned(refStrm));
+        }
```

Comment:
Do we have a similar test for signing ref assemblies with legacy provider?

Reply:
http://source.roslyn.io/#Roslyn.Compilers.CSharp.Emit.UnitTests/Emit/CompilationEmitTests.cs,41eb8fc53d084838